sonic-xcvrd/scripts/xcvrd

```diff
@@ -90,19 +93,17 @@ def log_error(msg, also_print_to_console=False):
 #========================== Signal Handling ==========================
 
 def signal_handler(sig, frame):
+    global RUN
```

Comment:
Kevin has submitted a PR to unify helper functions that used by all the daemons, signal handler is part of it since his PR is approved and above to merge, I think you need to look at his PR and align with him. Here is his PR: https://github.com/Azure/sonic-buildimage/pull/2570

Reply:
@keboliu i have reviewed Kevin's changes and i am strongly disagree with a such approach. I had a discussion with @andriymoroz-mlnx and decided to open this PR as is.

My major concerns are: 

1. Incorrect usage of signal handlers: throwing an exception in application main process thru calling sys.exit inside a signal handler is a bad practise
2. Placing a signal handlers in common class is also a bad practise

I think that each daemon should have a graceful shutdown flow.